src/libraries/System.Linq/src/System/Linq/Max.cs

```diff
@@ -465,6 +464,7 @@ public static decimal Max(this IEnumerable<decimal> source)
                     }
                     while (value == null);
 
+                    Comparer<TSource> comparer = Comparer<TSource>.Default;
```

Comment:
Out of curiosity, what is the motivation for not optimizing reference types?

Reply:
See issue #10050 for more details